soroban-env-host/src/host.rs

```diff
@@ -1588,6 +1588,35 @@ impl CheckedEnv for Host {
         Ok(self.add_host_object(sign_bytes.1)?.into())
     }
 
+    fn bigint_from_bytes_be(&self, sign: RawVal, bytes: Object) -> Result<Object, Self::Error> {
+        let s = self.bigint_sign_from_rawval(sign)?;
+        let res = self.visit_obj(bytes, |b: &Vec<u8>| {
+            MeteredBigInt::from_bytes_be(self.0.budget.clone(), s, b)
+        })?;
+        Ok(self.add_host_object(res)?.into())
+    }
+
+    fn bigint_from_radix_be(
+        &self,
+        sign: RawVal,
+        buf: Object,
+        radix: RawVal,
+    ) -> Result<Object, Self::Error> {
+        let s = self.bigint_sign_from_rawval(sign)?;
+        let r = self.bigint_radix_from_rawval(radix)?;
```

Comment:
I like the way this parsing happens with functions dedicated to the type of value being accepted. 🎉